contrail-ui/src/components/main/resource-list-view/ResourceListView.tsx

```diff
@@ -0,0 +1,200 @@
+
+    const emptyRows = rowsPerPage - Math.min(rowsPerPage, rows.length - page * rowsPerPage);
+  
+    const renderEmptyRows = emptyRows > 0 && (
+        <TableRow style={{ height: 49 * emptyRows }}>
+            <TableCell colSpan={6} />
+        </TableRow>
+    );
+
+    const renderResourceRows =
+        rows.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+            .map((row, index) => {
+                const isItemSelected = isSelected(row.resourceId);
+
+                function handleClickWrapper(event: React.MouseEvent<unknown>) {
+                    handleClick(event, row.resourceId);
+                }
+
+                return (
+                    <TableRow
+                        hover={true}
+                        onClick={handleClickWrapper}
+                        role="checkbox"
+                        tabIndex={-1}
+                        key={row.resourceId}
+                        selected={isItemSelected}
+                    >
+                        <TableCell padding="checkbox">
+                            <Checkbox
+                                checked={isItemSelected}
+                                color="default"
+                            />
+                        </TableCell>
+                        <TableCell align="left">{row.name}</TableCell>
+                        <TableCell align="left">{row.owner}</TableCell>
+                        <TableCell align="left">{row.dateCreated}</TableCell>
+                        <TableCell align="right">{row.fileSize}</TableCell>
+                    </TableRow>
+                );
+            });
+
+    function handleSelectAllClick(event: React.ChangeEvent<HTMLInputElement>) {
+        if (event.target.checked) {
+            const newSelecteds = rows.map(n => n.name);
+            setSelected(newSelecteds);
+            return;
+        }
+        setSelected([]);
+    }
+
+    function handleClick(event: React.MouseEvent<unknown>, name: string) {
+        const selectedIndex = selected.indexOf(name);
+        let newSelected: string[] = [];
+
+        if (selectedIndex === -1) {
+            newSelected = newSelected.concat(selected, name);
+        } else if (selectedIndex === 0) {
+            newSelected = newSelected.concat(selected.slice(1));
+        } else if (selectedIndex === selected.length - 1) {
+            newSelected = newSelected.concat(selected.slice(0, -1));
+        } else if (selectedIndex > 0) {
+            newSelected = newSelected.concat(
+                selected.slice(0, selectedIndex),
+                selected.slice(selectedIndex + 1),
+            );
+        }
+
+        setSelected(newSelected);
+    }
+
+    function handleChangePage(event: unknown, newPage: number) {
+        setPage(newPage);
+    }
+
+    function handleChangeRowsPerPage(event: React.ChangeEvent<HTMLInputElement>) {
+        setRowsPerPage(+event.target.value);
```

Comment:
what is this + sign

Reply:
This is easy way of converting string to number. I will use virgin way with parseInt

Reply:
ohh use >>0